src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/DllImportSearchPath.cs

```diff
@@ -4,6 +4,7 @@
 
 namespace System.Runtime.InteropServices
 {
+    // sync with src/mono/mono/metadata/native-library.c
```

Comment:
This is a public API - these values will never change. We can add new ones, but the need to revise both runtimes is pretty obvious.

Reply:
> > And you need to account for the linker removing the DefaultDllImportSearchPathsAttribute class (returning 0 is reasonable, I think).
> 
> CoreCLR generally checks attributes by doing a name match - if I declare a `System.Runtime.InteropServices.DefaultDllImportSearchPathsAttribute` in my own project for whatever odd reason, it's going to work the same as using the attribute from CoreLib.

I think @lambdageek was referring to this code having to work if a class of that name is fully missing at runtime. But we also use a name/namespace match to look for attributes in the Mono runtime, so this will work the same way as CoreCLR.